### DISCUSSION TITLE
[Feature] Track and Display "Upload Source (Device)" Metadata for Media

### DISCUSSION THREAD

**Discussion by SandiyosDev** [immich-app/immich#12012] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

This directly relates to the feature request [Mass remove already backed-up photos](https://github.com/immich-app/immich/issues/165), where users would like to mass delete on-device photos after they've been uploaded to Immich. Since the [Mobile upload refactor](https://github.com/immich-app/immich/issues/2567) is still ongoing, perhaps Immich can keep track of the sources of uploaded media—not based on the device they were taken with, according to their EXIF metadata, but based on the device/client/version that uploaded the particular asset.

This is the baseline requirement for implementing future features that require functionalities such as categorizing media, [identifying upload sources](https://old.reddit.com/r/immich/comments/1emfw9v/bulk_delete_by_source_whatsapp/), or downloading only the content that was previously uploaded by a specific device.

All in all, this feature request should be considered when implementing feature requests 165/2567, where the server would need to track media sources during uploads, with minor modifications needing to be made to the Web and Mobile clients.

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
#6906

**Reply by SandiyosDev** [reply]
Really? I suppose I didn't dig far enough to find this issue
